src/keygen/keyshare.rs

```diff
@@ -42,6 +49,80 @@ impl KeySharePrivate {
     pub(crate) fn public_share(&self) -> Result<CurvePoint> {
         CurvePoint::GENERATOR.multiply_by_scalar(&self.x)
     }
+
+    /// Convert private material into bytes.
+    ///
+    /// 🔒 This is intended for use by the calling application for secure
+    /// storage. The output of this function should be handled with care.
+    pub fn into_bytes(self) -> Vec<u8> {
+        // Format:
+        // KEYSHARE_TAG | key_len in bytes | key (big endian bytes)
+        //              | 8 bytes          | key_len bytes
+
+        let share = self.x.to_bytes();
+        let share_len = share.len().to_le_bytes();
+
+        [KEYSHARE_TAG, &share_len, &share].concat()
+    }
+
+    /// Convert bytes into private material.
+    ///
+    /// 🔒 This is intended for use by the calling application for secure
+    /// storage. Do not use this method to create arbitrary instances of
+    /// [`KeySharePrivate`].
+    pub fn try_from_bytes(bytes: Vec<u8>) -> Result<Self> {
+        // Expected format:
+        // KEYSHARE_TAG | key_len in bytes | key (big endian bytes)
+        //              | 8 bytes          | key_len bytes
+
+        let mut parser = ParseBytes::new(bytes);
+        let result = {
```

Comment:
In this function, we zeroize on the happy path at ` parser.zeroize();`. However, on error, we return early so we never have a chance to call `.zeroize()`. The simplest way I see to support this, would be to derive `ZeroizeOnDrop` for `ParseBytes`?

Reply:
Oh hmm. I thought the early returns in the result block would get captured by `result` so that we'd end up zeroizing in either case, but I added some extra logs and that doesn't seem to be the case.

It looks like my original idea would work if the result block was a function, like
```rust
let mut parse = || {
    ... all the parsing code
}
let result = parse();
parser.zeroize();
```

But that's kind of ugly; probably better to just `ZeroizeOnDrop`

Reply:
> thought the early returns in the result block would get captured by result

Sadly, no. `?` return out the current function, that's why wrapping it in a closure works!

The single log is nice. I agree this pattern is annoying!